src/sentry/api/endpoints/organization_events_vitals.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import absolute_import
+import sentry_sdk
+
+from rest_framework.response import Response
+from rest_framework.exceptions import ParseError
+
+from sentry.api.bases import OrganizationEventsV2EndpointBase, NoProjects
+from sentry.api.event_search import get_function_alias
+from sentry.snuba import discover
+
+
+class OrganizationEventsVitalsEndpoint(OrganizationEventsV2EndpointBase):
+    VITALS = {
+        "measurements.lcp": {"thresholds": [0, 2500, 4000]},
+        "measurements.fid": {"thresholds": [0, 100, 300]},
+        "measurements.cls": {"thresholds": [0, 0.1, 0.25]},
+        "measurements.fcp": {"thresholds": [0, 1000, 3000]},
+        "measurements.fp": {"thresholds": [0, 1000, 3000]},
+    }
+
+    def get(self, request, organization):
+        if not self.has_feature(organization, request):
+            return Response(status=404)
+
+        with sentry_sdk.start_span(op="discover.endpoint", description="parse params"):
+            try:
+                params = self.get_snuba_params(request, organization)
+            except NoProjects:
+                return Response([])
+
+            vitals = [vital.lower() for vital in request.GET.getlist("vital", [])]
+            if len(vitals) == 0:
+                raise ParseError(detail=u"Need to pass at least one vital")
+
+            selected_columns = []
+            aliases = {}
+            for vital in vitals:
+                if vital not in self.VITALS:
+                    raise ParseError(detail=u"{} is not a valid vital".format(vital))
+                aliases[vital] = []
+                for index, threshold in enumerate(self.VITALS[vital]["thresholds"]):
+                    column = f"count_at_least({vital}, {threshold})"
```

Comment:
Would having this validation and conversion code live in a serializer help make it simpler?

Reply:
I think for now I'd rather keep this here cause its only validating that the vitals are real. But if there's a need to validate more I think it'd be worth moving this to a serializer.